node/src/types/chainspec/network_config.rs

```diff
@@ -0,0 +1,145 @@
+use std::path::Path;
+
+use csv::ReaderBuilder;
+use datasize::DataSize;
+#[cfg(test)]
+use rand::Rng;
+use serde::{Deserialize, Serialize};
+
+use casper_execution_engine::{core::engine_state::genesis::GenesisAccount, shared::motes::Motes};
+use casper_types::{
+    bytesrepr::{self, FromBytes, ToBytes},
+    PublicKey, U512,
+};
+
+use super::error::ChainspecAccountsLoadError;
+#[cfg(test)]
+use crate::testing::TestRng;
+use crate::types::Timestamp;
+
+const CHAINSPEC_ACCOUNTS_FILENAME: &str = "accounts.csv";
+
+/// This parses the accounts.csv file if it exists.
+///
+/// Returns `Ok` if there is no such file, or if there is and it can be parsed correctly.  Returns
+/// an error if the file exists but can't be parsed.
+pub(super) fn parse_accounts_csv<P: AsRef<Path>>(
+    dir_path: P,
+) -> Result<Vec<GenesisAccount>, ChainspecAccountsLoadError> {
+    #[derive(Deserialize)]
+    struct ParsedAccount {
+        public_key: PublicKey,
+        balance: U512,
+        bonded_amount: U512,
+    }
+
+    let path = dir_path.as_ref().join(CHAINSPEC_ACCOUNTS_FILENAME);
+    if !path.is_file() {
+        return Ok(vec![]);
```

Comment:
I am not sure it's safe to treat an empty set of Genesis validators as an OK variant. This will panic at `assert!` in `EraSupervisor` with `"cannot start era with total weight 0"` error message. I think it's better to fail early.

Reply:
I generally agree about failing early, but from the chainspec perspective, it doesn't and shouldn't care whether the list is empty or not.  The idea is that we can easily introduce a different `accounts.csv` at a future version if we see a need.

I think such a check should be outside of the chainspec/chainspec_loader since it doesn't really know the context in which it's running (i.e. whether this is a genesis run or a subsequent version where currently chainspec_accounts would be ignored)

Reply:
I had the same thoughts initially but I think that it's fine to not have the `accounts.csv` file present – in which case we don't overwrite the validator set, but it's not OK to have `accounts.csv` file that is empty.

Reply:
> but it's not OK to have accounts.csv file that is empty.

Hmm - that seemed like a good point at first, but the more I think on it, the less sure I am.  Could there be a use-case for an empty accounts.csv?  It _is_ a valid file after all - it can be parsed into a `Vec<GenesisAccount>`.

Ultimately it shouldn't be a problem as we can always change the behaviour in a future update.  I'd like to hear what others think though, but I'm not against your suggestion.

Reply:
> Could there be a use-case for an empty `accounts.csv`?

Not with the code we have now, at least not on the consensus side. Not sure how it's interpreted in the EE.